R/eval-tidy.R

```diff
@@ -342,3 +339,12 @@ f_self_eval <- function(overscope, overscope_top) {
     .Call(rlang_eval, f_rhs(f), overscope)
   }
 }
+tilde_eval <- function(f, env) {
+  # Inline the base primitive because overscopes override `~` to make
+  # quosures self-evaluate
+  mut_node_car(f, base::`~`)
+
+  # Change both old and new formulas to original head avoid side-effects
+  on.exit(mut_node_car(f, sym_tilde))
```

Comment:
hmm that, probably not, but we do manipulate calls in place. Most (all?) of the time after duplication though.

The compiler doesn't do anything to tilde calls currently, and I don't think it ever will because it would need runtime information to do so. If you prefer I can just duplicate the tilde call so we're sure.